### PR TITLE
CSP-165-web-UpdateProviderLocationDetails

### PR DESCRIPTION
**Pull request by raviudari01**

### CODE REVIEW

File: src/SFA.DAS.Roatp.CourseManagement.Web/Controllers/ViewProviderLocationDetailsController.cs [hunk 1]
```diff
@@ -43,6 +43,7 @@ public async Task<IActionResult> GetProviderLocationDetails([FromRoute] Guid Id)
 
             model = (ProviderLocationViewModel)result.ProviderLocation;
             model.BackUrl = Url.RouteUrl(RouteNames.GetProviderLocations, new { ukprn = Ukprn });
+            model.UpdateContactDetailsUrl = Url.RouteUrl(RouteNames.GetUpdateProviderLocationDetails, new { ukprn = Ukprn, Id });
```
Comment (MarkFCain) [inline src/SFA.DAS.Roatp.CourseManagement.Web/Controllers/ViewProviderLocationDetailsController.cs:46]: Can you add coverage for this?

File: src/SFA.DAS.Roatp.CourseManagement.Web.UnitTests/Controllers/EditProviderLocationDetailsControllerTests/EditProviderLocationDetailsControllerGetTests.cs [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+                .ReturnsAsync(queryResult);
+
+            var result = await _sut.GetProviderLocationDetails(id);
+
+            var viewResult = result as ViewResult;
+            viewResult.Should().NotBeNull();
+            viewResult.ViewName.Should().Contain("EditProviderLocationsDetails.cshtml");
+            var model = viewResult.Model as ProviderLocationViewModel;
+            model.Should().NotBeNull();
+            model.BackUrl.Should().NotBeNull();
```
Comment (MarkFCain) [inline src/SFA.DAS.Roatp.CourseManagement.Web.UnitTests/Controllers/EditProviderLocationDetailsControllerTests/EditProviderLocationDetailsControllerGetTests.cs:53]: add check for CancelUrl

File: src/SFA.DAS.Roatp.CourseManagement.Web.UnitTests/Controllers/EditProviderLocationDetailsControllerTests/EditProviderLocationDetailsControllerGetTests.cs [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+                .ReturnsAsync((GetProviderLocationDetailsQueryResult)null);
+
+            var result = await _sut.GetProviderLocationDetails(id);
+
+            var viewResult = result as ViewResult;
+            viewResult.Should().NotBeNull();
+            viewResult.ViewName.Should().Contain("EditProviderLocationsDetails.cshtml");
+            var model = viewResult.Model as ProviderLocationViewModel;
+            model.Should().NotBeNull();
+            model.BackUrl.Should().NotBeNull();
```
Comment (MarkFCain) [inline src/SFA.DAS.Roatp.CourseManagement.Web.UnitTests/Controllers/EditProviderLocationDetailsControllerTests/EditProviderLocationDetailsControllerGetTests.cs:70]: Add check for CancelUrl

**Review by MarkFCain — APPROVED** [review]
minor requests for coverage

File: src/SFA.DAS.Roatp.CourseManagement.Web/Controllers/EditProviderLocationDetailsController.cs [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+            var model = (ProviderLocationViewModel)result.ProviderLocation;
+            model.BackUrl = model.CancelUrl = Url.RouteUrl(RouteNames.GetProviderLocations, new { ukprn = Ukprn });
+            return model;
+        }
+
+        [Route("/providers/{ukprn}/locations/{Id}/update-location-details", Name = RouteNames.PostUpdateProviderLocationDetails)]
+        [HttpPost]
+        public async Task<IActionResult> UpdateProviderLocationDetails(ProviderLocationDetailsSubmitModel model, [FromRoute] Guid Id)
+        {
+           if (ModelState.IsValid) await CheckIfNameIsAvailable(model.LocationName, Id);
```
Comment (DevenKShah) [inline src/SFA.DAS.Roatp.CourseManagement.Web/Controllers/EditProviderLocationDetailsController.cs:61]: Alignment

**Review by DevenKShah — APPROVED** [review]

**Pull request merged**